forem/forem
# Connect: Highlight @ mentions to logged in user

## What type of PR is this? (check all applicable)


## Description
This PR distinguishes between mentions to the logged in user and mentions to other users by wrapping @-mentions to the logged in user with `<mark>`.  

Based on my understanding, it's not possible to wrap `<mark>` tags at the time the message's HTML is generated in `models/message.rb`. Because of this, we'll have to highlight the @-mentions on the client side.

The implementation in this PR makes this feature quite obscure but it happens to be the implementation with the fewest lines of changed code, so I just rolled with it.

## Related Tickets & Documents
#9472 

## QA Instructions, Screenshots, Recordings
1. Join a group chat
2. Mention someone else, observe that their link is not highlighted.
3. Mention yourself, observe that your link is wrapped with `<mark>`  

![Screenshot from 2020-09-04 21-39-33](https://user-images.githubusercontent.com/5664347/92297737-272fd380-eef7-11ea-83a1-e31a78a6e25b.png)


## Added tests?


## Added to documentation?
